src/06-positron-rag/lib/fetchArticles.ts

```diff
@@ -0,0 +1,99 @@
+/* eslint-disable @typescript-eslint/no-explicit-any */
+import fetch from "node-fetch"
+
+export async function fetchArticles(
+  minArticleCount: number = 10,
+  batchSize: number = 10,
+  sleepMs: number = 1000
+) {
+  const url = "https://metaphysics-staging.artsy.net/v2"
+  let afterCursor = ""
+  let articles: any[] = []
+
+  while (articles.length < minArticleCount) {
+    const query = `
+    {
+      articlesConnection(first: ${batchSize}, after: "${afterCursor}", sort: PUBLISHED_AT_DESC) {
+        pageInfo {
+          hasNextPage
+          endCursor
+        }
+        edges {
+          cursor
+          node {
+            # meta
+            internalID
+            title
+            searchTitle # unused?
+            href
+            publishedAt
+
+            # creators
+            byline
+            authors {
+              name
+            }
+
+            # summary
+            description
+            keywords
+            leadParagraph # unused?
+
+            # buckets
+            channel  {
+              internalID
+              slug
+              name
+              type
+            }
+            vertical
+            seriesArticle {
+              href
+            }
+
+            # content
+            sections {
+              __typename
+              ... on ArticleSectionText {
+                body
+              }
+            }
+          }
+        }
+      }
+    }
+    `
+
+    try {
+      const response = await fetch(url, {
+        method: "POST",
+        headers: { "Content-Type": "application/json" },
+        body: JSON.stringify({ query }),
+      })
+
+      process.stdout.write(".")
+
+      const data: any = await response.json()
+      articles = articles.concat(
+        data.data.articlesConnection.edges.map((edge: any) => edge.node)
+      )
+
+      await new Promise<void>((resolve) => {
+        setTimeout(() => {
+          resolve()
+        }, sleepMs)
+      })
```

Comment:
If I'm reading this right we're just pausing for a second. Any specific reason why?

Reply:
Just trying to avoid self DDoS 😅 

This by default pauses for 1 sec between each batch of 10. Both nums are configurable.